core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -811,8 +818,38 @@ private[spark] class TaskSchedulerImpl(
     blacklistTrackerOpt.map(_.nodeBlacklist()).getOrElse(scala.collection.immutable.Set())
   }
 
+  // Add a on-off switch to save time for rack resolving
+  private def skipRackResolving: Boolean = sc.conf.get(LOCALITY_WAIT_RACK) == 0L
```

Comment:
I was just reviewing another patch related to delay scheduling, and I realized this optimization is a bit too aggressive.  That configuration only controls whether you *wait* for a resource that is rack-local.  Even when the wait is 0, spark still tries to to find a rack-local task for a given resource offer; it just will schedule a non-rack-local task even if it can't find a rack-local one.  But it won't be able to do that if it doesn't know what racks the resource offers are on.  

So I think you either need to:

a) change this to use a new conf, with an extra check that you only turn off rack resolution entirely if its *also* true that `sc.conf.get(LOCALITY_WAIT_RACK) == 0L`

b) is this optimization even needed, considering how much time the rest of this change should save?  Maybe we should still always do the rack resolution, since it should be pretty fast after the rest of your change.

Reply:
I knew that. Isolation deployment of computing and storage is more and more popular in industry. If someone set `LOCALITY_WAIT` to 0 on purpose, almost time it is no need data locality (especially rack level) at all. Rack resolving and locality algorithm still spends time in no need data locality case. Could we open a new ticket to address this isolation case?

Reply:
@squito [SPARK-27232](https://issues.apache.org/jira/browse/SPARK-27232) also faces same situation

Reply:
thanks for pointing out the other pr that is similar -- yes I agree, I can see cases where you would want to skip rack resolution entirely for the "disaggregated" clusters you are talking about.  I'll comment on the other ticket as well.  Would you like to follow up with that?